### PR TITLE
fix: [BiW Editor] Disable objects culling

### LINKED ISSUE #519: BiW set Name and Description while Publishing
Players should be able to set a name and description of the scene that they are about to publish during the publishing process.

When players push the publish button: 
- [x] A new Modal will appear
- [x] It will contain two inputs of Name and Description
- [x] Name Input should be one line
- [x] Description input should be multiline
- [x] Description should be optional
- [x] Initial information should be retrieved from the Project Scene
- [x] If no information is retreived, default should be "My new place" for the name, and empty for the description
- [x] The modal will also contain a Publish and Cancel button.
- [x] After clicking on Publish, the flow will continue as normal, starting from the confirmation dialog.
- [x] This information should be pushed to the Project Scene through the builder API and to the Content Server

* This is not final design and needs to be supervised

### LINKED ISSUE #524: BiW custom asset packs compatibility
Players are currently able to create Custom Asset packs using the original Builder. In this asset packs, they define a screenshot, a name, and can import new Assets form ZIP, GLB/ GLTF or FBX files. 

While we do not intend to replicate this functionality inside BiW for the time being, we want to have compatibility to those assets by displaying it and make them usable in BiW.

- [x] If a player creates a new Custom Asset Pack, that new category needs to appear in Asset Packs inside BiW
- [x] That asset pack will have the screenshot defined in the original Builder.
- [x] Asset inside this collections will be usable and searchable like any other asset.
- [ ] If a project contain assets that belongs to a missing asset pack, those instances will be replace by a MISSING asset: a PINK BOX with name "Missing Asset #"

### LINKED ISSUE #619: BiW sometimes editor does not load asset in the Scene
Sometimes, when you enter to edit a Land that has an associated Project, the assets (3D Models) do not load until you add another entity to the scene.

The circumstances to reproduce this situation are unknown.

### LINKED ISSUE #624: BiW: disable objects culling
In scenes bigger than one parcel sometimes objects are not shown due to the culling

### PR DESCRIPTION
**Pull request by sandrade-dcl**
### Fixes #624 
- [x] Disable objects culling during the edition.

### Fixes #619 
- [x] Sometimes, when you enter to edit a Land that has an associated Project, the assets (3D Models) do not load until you add another entity to the scene.

## How to test it?

1. Go to: https://play.decentraland.zone/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:fix/BiW-Disable-objects-culling&ENV=zone&ENABLE_BUILDER_IN_WORLD&position=80,154
2. Press 'K' to open the Builder In World editor.
3. Notice all the entities are visible.
4. Exit from the editor.
5. Enter back into the editor.
6. Notice all the entities are visible.
7. Repeat the process several times.

## Annotations
Finally I had not have to use the new implemented class `SettingsPanelDataStore` for this issue, but I would like to keep it in the project because I think it is something useful that we're going to use in the future. This data store class let us to change any setting control from any place of our code without the necessity to go and click fisically in the control inside the SettingsPanelHUD.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by Suduck — APPROVED** [review]
LGTM!

**Review by BrianAmadori — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: As I left indicated in the PR "annotations", finally I had not to use the `SettingsPanelDataStore` class, so we could remove it if we want.

`SettingsPanelDataStore.i.controls.controlControllers` would be only a collection of controllers that are initially added when the own control is added in the HUD. The initial idea of this was to have some kind of accessor to all our settings controls (only to its controllers) in order to be able to modify them from code without having to go to the HUD panel to physically click on the control. I thought it could be very useful for cases where we need to apply any setting from code and, apart of apply this setting internally, to see visually applied the change in the HUD.

For example: Imagine that in some part of our code we need to change the bloom control setting. We could do this:
`SettingsPanelDataStore.i.TryGetController(out BloomControlController bloomController);`
`bloomController.UpdateSetting(<new value>);`

In this way, the bloom would be changed and we would also see the change reflected in the corresponding settings HUD toogle.

What would be your concern about this kind of solution?

Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: I dont like this approach neither, we are trying to move to a Data Driven architecture and minimize the use of singletons.

Why cannot we move the settings model to a new assembly to be referenced by the `DataStore` and use it? We can even have a subclass in the `DataStore` with each setting on its own.

If you wanna change a setting, you go to the DataStore, change it and the systems that need to know about that variable will subscribe to its `OnChange` and react properly.

Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: @AjimenezDCL but this is what I exactly doing with this approach, but instead using our general data store, I'm using a specific data store for the settings panel.

Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: Anyway guys, if you don't mind, I would prefer to move this discussion to a Post-Weekly and avoid to block this PR. It is something that was implemented in an initial point of this issue but that I have not used later. I simply left the class there for future uses.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs:765]: I think I already tried on the #663 and I had the same problem. anyway let me confirm and I will tell you 👍

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: Bear in mind that with the new environment assembly, `DataStore` will stop being a singleton and will become a reference, so you would have to include this `SettingsPanelDataStore` in `Environment` somehow. 

This approach also have the downside that if you want to change a setting you have to reference the whole `SettingsHUD` assembly, which seems strange.

Moving on and continuing the discussion in a post-daily/weekly it's fine by me.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs:765]: Tested with the #663 and I still have the same problem. I have also tested different approach to avoid this delay hack but nothing that fix it for now. If you don't mind @BrianAmadori, could we have a quick meet to review the culling flow? You may see something I don't.

Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs]: After the pair programming with @BrianAmadori, we have finally solved the problem! Changes done in https://github.com/decentraland/unity-renderer/pull/650/commits/1b7bce84d84edf87423283d8f6c12f4a5bbc8473 and https://github.com/decentraland/unity-renderer/pull/650/commits/19c0c67fbe32f3345b4ba06b0f47ccc5f0f6236b. Thanks Brian!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs
Comment (BrianAmadori) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs:42]: Agree with @AjimenezDCL and move this discussion for later. Ty for the explanation!

File: unity-renderer/Assets/Rendering/Culling/Tests/CullingControllerAndScenesShould.cs
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/Culling/Tests/CullingControllerAndScenesShould.cs:107]:
```suggestion
```

We should immediately check if the entity was brought back. As the bug was that Stop() is not enabling the culling setting of disabled gameObjects.

**Review by BrianAmadori — APPROVED** [review]
Pre-approving with a small request. Please address this before merging. I'm really glad we managed to fix this properly. 🎉

**Pull request merged**
